spec/controllers/api/submissions_controller_spec.rb

```diff
@@ -20,6 +22,13 @@
         it { is_expected.to include 'message' }
         it { is_expected.to include 'result' }
       end
+
+      describe 'when sent invalid data from the public' do
```

Comment:
Reading this I realised that this controller spec is more of a feature test, rather than controller spec. And then I didn't find a feature test, so I looked into it a bit more. 

I think we definitely should have a feature (request) spec to handle various cases, which are end-to-end. Including authentication problems, input errors, etc. 
The controller spec is theoretically only concerned about the various objects to call, rather then inputs and outputs. Hope that makes sense. 

I think we can go with this, though and look into the request testing as a separate bit, if you want?